Benqi-Agents/Large-Delegate-Votes-Change/src/agent.spec.ts

```diff
@@ -0,0 +1,122 @@
+
+      const findings = await handleTransaction(txEvent);
+
+      expect(findings).toStrictEqual([]);
+    });
+
+    it("should ignore DelegateVotesChanged events from other contracts", async () => {
+      const alertLog = QI_IFACE.encodeEventLog(QI_IFACE.getEvent("DelegateVotesChanged"), [
+        DELEGATE_ADDRESS, // delegate
+        "100", // previousBalance
+        increaseByPercentage("100", PERCENTAGE_THRESHOLD), // newBalance
+      ]);
+
+      const txEvent = new TestTransactionEvent();
+      txEvent.addAnonymousEventLog(createAddress("0xd"), alertLog.data, ...alertLog.topics);
+
+      const findings = await handleTransaction(txEvent);
+
+      expect(findings).toStrictEqual([]);
+    });
+
+    it("should ignore DelegateVotesChanged events from the QI contract that do not meet the threshold", async () => {
+      const alertLog = QI_IFACE.encodeEventLog(QI_IFACE.getEvent("DelegateVotesChanged"), [
+        DELEGATE_ADDRESS, // delegate
+        "100", // previousBalance
+        increaseByPercentage("100", PERCENTAGE_THRESHOLD), // newBalance
+      ]);
+
+      const ignoredLog = QI_IFACE.encodeEventLog(QI_IFACE.getEvent("DelegateVotesChanged"), [
+        DELEGATE_ADDRESS, // delegate
+        "100", // previousBalance
+        increaseByPercentage("100", PERCENTAGE_THRESHOLD.sub(1)), // newBalance
+      ]);
+
+      const txEvent = new TestTransactionEvent();
+      txEvent.addAnonymousEventLog(QI_ADDRESS, alertLog.data, ...alertLog.topics);
+      txEvent.addAnonymousEventLog(QI_ADDRESS, ignoredLog.data, ...ignoredLog.topics);
+
+      const findings = await handleTransaction(txEvent);
+
+      expect(findings).toStrictEqual([
+        createFinding(DELEGATE_ADDRESS, "100", increaseByPercentage("100", PERCENTAGE_THRESHOLD)),
+      ]);
+    });
+
+    it("should detect multiple eligible events from the QI contract", async () => {
+      const alertLog1 = QI_IFACE.encodeEventLog(QI_IFACE.getEvent("DelegateVotesChanged"), [
+        DELEGATE_ADDRESS, // delegate
+        "100", // previousBalance
+        increaseByPercentage("100", PERCENTAGE_THRESHOLD), // newBalance
+      ]);
+
+      const alertLog2 = QI_IFACE.encodeEventLog(QI_IFACE.getEvent("DelegateVotesChanged"), [
+        DELEGATE_ADDRESS, // delegate
+        "200", // previousBalance
+        increaseByPercentage("200", PERCENTAGE_THRESHOLD), // newBalance
+      ]);
+
+      const txEvent = new TestTransactionEvent();
+      txEvent.addAnonymousEventLog(QI_ADDRESS, alertLog1.data, ...alertLog1.topics);
+      txEvent.addAnonymousEventLog(QI_ADDRESS, alertLog2.data, ...alertLog2.topics);
+
+      const findings = await handleTransaction(txEvent);
+
+      expect(findings).toStrictEqual([
+        createFinding(DELEGATE_ADDRESS, "100", increaseByPercentage("100", PERCENTAGE_THRESHOLD)),
+        createFinding(DELEGATE_ADDRESS, "200", increaseByPercentage("200", PERCENTAGE_THRESHOLD)),
+      ]);
+    });
+  });
```

Comment:
To avoid having all the positive test cases with an increase always equal to the threshold, You can make some with a greater value (`PERCENTAGE_THRESHOLD + 10` for example)